### PR TITLE
Add Microsoft Entra ID administrator management for Azure SQL servers

### PR DESCRIPTION
**Pull request by jongio**
Implement a comprehensive command to retrieve and list Microsoft Entra ID administrators for Azure SQL servers. This PR adds the `azmcp sql server entraadmin list` command with full Azure SDK integration, error handling, and comprehensive testing.

## Key Features
- **New Command**: `azmcp sql server entraadmin list` - Lists all Microsoft Entra ID administrators for a specified SQL server
- **Full Entra ID Support**: Uses proper Microsoft Entra ID terminology throughout (updated from legacy Azure Active Directory naming)
- **Azure SDK Integration**: Leverages Azure.ResourceManager.Sql for reliable administrator retrieval
- **Comprehensive Testing**: Includes both unit tests and live integration tests
- **Complete Documentation**: Updated README, command docs, and developer guides

## Changes Made
- Added new `EntraAdminListCommand` with comprehensive validation and error handling
- Created `SqlServerEntraAdministrator` model with proper JSON serialization
- Updated `ISqlService` interface and implementation with `GetEntraAdministratorsAsync`
- Added complete test coverage including edge cases and error scenarios
- Reorganized documentation alphabetically for improved navigation
- Updated terminology throughout codebase from "Active Directory" to "Microsoft Entra ID"

## Breaking Changes
None - this is a new feature addition that doesn't affect existing functionality.

## Testing
- ✅ Unit tests with full code coverage
- ✅ Live integration tests with Azure resources
- ✅ Parameter validation and error handling tests
- ✅ JSON serialization validation

### CODE REVIEW

File: src/Areas/Sql/Commands/AdAdmin/AdAdminListCommand.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+public sealed class AdAdminListCommand(ILogger<AdAdminListCommand> logger)
+    : BaseSqlCommand<AdAdminListOptions>(logger)
+{
+    private const string CommandTitle = "List SQL Server AD Administrators";
+
+    public override string Name => "list";
+
+    public override string Description =>
+        """
+        Gets a list of Azure Active Directory administrators for a SQL server. This command retrieves all
```
Comment (xiangyan99) [inline src/Areas/Sql/Commands/AdAdmin/AdAdminListCommand.cs:21]: Isn't Azure Active Directory renamed to Entra ID?

Comment (ericshape) [reply src/Areas/Sql/Commands/AdAdmin/AdAdminListCommand.cs:21]: yes

**Review by vcolin7 — APPROVED** [review]

File: src/Areas/Sql/Commands/AdAdmin/AdAdminListCommand.cs (hunk 1 above)
Comment (AV25242) [reply src/Areas/Sql/Commands/AdAdmin/AdAdminListCommand.cs:21]: @xiangyan99 and @wbreza yes it is however `az cli` still refers to it as `ad-admin` and thats where we went that route, but I hear you out. The portal calls it Entra admin so I think its fair to call it entra.

@jongio how difficult will it be to change it to `azmcp  sql server` **entra** `-admin list`

let me know, I will update the spec as well.

Comment (jongio) [reply src/Areas/Sql/Commands/AdAdmin/AdAdminListCommand.cs:21]: We can't have dashes in the name, so it would be entraadmin list.

File: docs/azmcp-commands.md [hunk 2]
```diff
@@ -505,8 +505,8 @@ azmcp sql db show --subscription <subscription> \
 
 ### Azure SQL Server Operations
 ```bash
-# List Active Directory administrators for a SQL server
-azmcp sql server adadmin list --subscription <subscription> \
+# List Microsoft Entra ID administrators for a SQL server
+azmcp sql server entraadmin list --subscription <subscription> \
```
Comment (vcolin7) [inline docs/azmcp-commands.md:509]: Is this preferred over `entraidadmin`? I don't hold a strong opinion either way, I just noticed we replaced AD with Entra ID in a bunch of docs but replaced `ad` with `entra` in the command and related classes.

Comment (jongio) [reply docs/azmcp-commands.md:509]: @AV25242 ?

Comment (AV25242) [reply docs/azmcp-commands.md:509]: @ericshape here is my thought , below is how its displayed in Azure SQL UI, ideally I would have preferred entra-admin because its not any id but an admin, however with the limitation of not being able to have a hyphen I am ok with entraadmin as its truly administrator at this point

image
